pkg/storage/redis.go

```diff
@@ -60,6 +63,47 @@ func (b *RedisDB) Init(i interface{}) error {
 	return nil
 }
 
+func processRedisOptions(opts ...Option) (address, password string, err error) {
+	if len(opts) != 2 {
+		return "", "", errors.New("redis options must contain address and password")
+	}
+	for _, opt := range opts {
+		switch opt.ID {
+		case RedisAddressOption:
+			maybeAddress, ok := opt.Option.(string)
+			if !ok {
+				err = errors.New("redis address must be a string")
+				return
+			}
+			if len(maybeAddress) == 0 {
+				err = errors.New("redis address must not be empty")
+				return
+			}
+			address = maybeAddress
+		case PasswordOption:
+			maybePassword, ok := opt.Option.(string)
+			if !ok {
+				err = errors.New("redis password must be a string")
+				return
+			}
+			if len(maybePassword) == 0 {
+				err = errors.New("redis password must not be empty")
+				return
+			}
+			password = maybePassword
+		}
+	}
+	if len(address) == 0 {
+		err = errors.New("redis address must not be empty")
+		return
+	}
+	if len(password) == 0 {
+		err = errors.New("redis password must not be empty")
+		return
+	}
```

Comment:
Seem repeated. Is it possible to DRY?

Reply:
condensed this into a single if. need to check whether the cases were hit for required options, so think this is as simple as it can be